.gitattributes

```diff
@@ -11,3 +11,9 @@ system/ linguist-vendored
 
 AUTHORS merge=union
 ChangeLog.md merge=union
+
+# line endings
+* text=auto eol=lf
```

Comment:
Just to be clear, this means that for almost all files, developers will see unix line endings, even on windows?  Why is that desirable?

Reply:
Any executable file that is supposed to run on Linux must be lf, and any executable that wants to run on Windows must be crlf. 
If someone clones this repository on Windows and then opens it in WSL, the line-endings will break because they become CRLF by default if no option is specified.

So, to fix these issues, one convention should be chosen. The easiest one is to specify CRLF for Windows executables and make everything else LF.

Reply:
If this is just to support WSL users who are using a kind of hybrid environment I'm not sure it worth it.   I would have though that WSL users already have to savy enough to setup global settings.

In any case, can we split out this PR so we can discuss the `.gitattributes` separately.  It seems like its worth further discussion and I don't want to block to this PR.